R/verb-distinct.R

```diff
@@ -173,6 +198,11 @@ add_distinct <- function(.data) {
     return(out)
   }
 
-  lazy_query$distinct <- TRUE
+  if (!is.logical(distinct)) {
+    lazy_query$distinct <- new_lazy_select(distinct)
```

Comment:
Echoing one of my earlier comments, I think things might be simpler if you did the conversion to `new_lazy_select()` later.